### PR TITLE
Add documentation for task timeouts

### PR DESCRIPTION
**Pull request by benbp**
Depends on https://github.com/RackHD/on-tasks/pull/184

@RackHD/corecommitters 
@heckj I lied, there actually wasn't any documentation for this before!

### CODE REVIEW

**Comment by anhou** [comment]
:+1:

File: docs/rackhd/tasks.rst [hunk 1]
```diff
@@ -408,6 +405,43 @@ If a user overrides ``deleteAll`` to be false, and ``raidIds`` to be ``[0,1,2]``
...
+
+In the task options object, a magic value `$taskTimeout` can be used to specify a maximum
+amount of time a task may be run, in milliseconds. By default, this value is equal to 24 hours.
+To specify an infinite timeout, a value of 0 or -1 may be used.
+
+.. code-block:: js
+
+    {
+        "options": {
+            "$taskTimeout": 3600000  // 1 hour timeout (in ms)
```
Comment (yyscamper) [inline docs/rackhd/tasks.rst]: Lack the closing `}`

File: docs/rackhd/tasks.rst [hunk 2]
```diff
@@ -408,6 +405,43 @@ If a user overrides ``deleteAll`` to be false, and ``raidIds`` to be ``[0,1,2]``
...
+    }
+
+.. code-block:: js
+
+    {
+        "options": {
+            "$taskTimeout": -1  // no timeout
```
Comment (yyscamper) [inline docs/rackhd/tasks.rst]: Lack the closing `}`

File: docs/rackhd/tasks.rst [hunk 3]
```diff
@@ -408,6 +405,43 @@ If a user overrides ``deleteAll`` to be false, and ``raidIds`` to be ``[0,1,2]``
...
+
+For backwards compatibility reasons, task timeouts can also be specified via the `schedulerOverriddes` option:
+
+.. code-block:: js
+
+    {
+        "options": {
+            "schedulerOverrides": {
+                "timeout": 3600000
+            }
```
Comment (yyscamper) [inline docs/rackhd/tasks.rst]: Lack a closing `}`

**Comment by yyscamper** [comment]
:after these `{` are added

**Comment by benbp** [comment]
Whoops, thanks @yyscamper, updated.

**Pull request merged**